spyderlib/restart_app.py

```diff
@@ -126,21 +232,63 @@ def main():
     command = '"{0}" "{1}" {2}'.format(python, spyder, args)
 
     # Adjust the command and/or arguments to subprocess depending on the OS
-    shell = os.name != 'nt'
+    shell = not IS_WINDOWS
 
-    # Wait for original process to end before launching the new instance
-    while True:
+    # Before launching a new Spyder instance we need to make sure that the
+    # previous one has closed. We wait for a fixed and "reasonable" amount of
+    # time and check, otherwise an error is launched
+    wait_time = 60*5 if IS_WINDOWS else 60*3 # Seconds
+    for counter in range(int(wait_time/SLEEP_TIME)):
         if not is_pid_running(pid):
             break
-        time.sleep(0.2)  # Throttling control
+        time.sleep(SLEEP_TIME)  # Throttling control
+        QApplication.processEvents()  # Needed to refresh the splash
+    else:
+        # The old spyder instance took too long to close and restart aborts
+        restarter.launch_error_message(error_type=CLOSE_ERROR)
 
     env = os.environ.copy()
+
+    # Reset Spyder (if required)
+    # -------------------------------------------------------------------------
+    if reset:
+        restarter.set_splash_message(_('Resetting Spyder to defaults'))
+        command_reset = '"{0}" "{1}" {2}'.format(python, spyder, args_reset)
+
+        try:
+            p = subprocess.Popen(command_reset, shell=shell, env=env)
+        except Exception as error:
+            restarter.launch_error_message(error_type=RESET_ERROR, error=error)
+        else:
+            p.communicate()
+            pid_reset = p.pid
+
+        # Before launching a new Spyder instance we need to make sure that the
+        # reset subprocess has closed. We wait for a fixed and "reasonable"
+        # amount of time and check, otherwise an error is launched.
+        wait_time = 60  # Seconds

```

Comment:
This is the reset process, the reset process normally is fast on windows as well. Is on waiting or the spyder instance to close that takes long, [check this](https://github.com/spyder-ide/spyder/pull/2423/files#diff-6614a100b3eb58a6c02b7b1da9e72fa1R240)